MacrosTranslator/src/translator/commands/commonCommands/unique_commands.h

```diff
@@ -0,0 +1,282 @@
+        return result.str();
+    };
+};
+
+class ScnCiteCommand : public ScSCnTexCommand
+{
+public:
+    std::string Complete(TexCommandParams const &params) override
+    {
+        std::stringstream result;
+        result << texSlash << "cite" << openBr << GetFirstParameter(params) << closeBr;
+        return result.str();
+    };
+};
+
+class IncludeGraphicsCommand : public ScSCnTexCommand
+{
+public:
+    std::string Complete(TexCommandParams const &params) override
+    {
+        std::stringstream result;
+        result << "[20em]" << openBr << GetLastParameter(params) << closeBr;
+        return result.str();
+    };
+};
+
+class ScnFileImageCommand : public ScSCnTexCommand
+{
+public:
+    std::string Complete(TexCommandParams const &params) override
+    {
+        std::stringstream result;
+        result << texSlash << "scnfileimage" << openBr << GetFirstParameter(params) << closeBr << newline;
+        return result.str();
+    };
+};
+
+class ScnFileScgCommand : public ScSCnTexCommand
+{
+public:
+    std::string Complete(TexCommandParams const &params) override
+    {
+        std::stringstream result;
+        result << texSlash << "scnfileimage[20em]" << openBr << GetFirstParameter(params) << closeBr << newline;
+        return result.str();
+    };
+};
+
+class ScnEqScgCommand : public ScSCnTexCommand
+{
+public:
+    std::string Complete(TexCommandParams const &params) override
+    {
+        std::stringstream result;
+        result << texSlash << "scneq" << openBr << texSlash << "scnfileimage[20em]" << openBr
+               << GetFirstParameter(params) << closeBr << closeBr << newline;
+        return result.str();
+    };
+};
+
+class ScnHasElementsCommand : public ScSCnTexCommand
+{
+public:
+    std::string Complete(TexCommandParams const &params) override
+    {
+        std::stringstream result;
+        result << texSlash << "begin" << openBr << "scnhaselementrolelist" << closeBr << newline << ParseByDelim(params)
+               << texSlash << "end" << openBr << "scnhaselementrolelist" << closeBr << newline;
+        return result.str();
+    };
+};
+
+class ScnSdRelationCommand : public ScSCnTexCommand
+{
+public:
+    std::string Complete(TexCommandParams const &params) override
+    {
+        std::stringstream result;
+        result << texSlash << "begin" << openBr << "scnhaselementrolelist" << closeBr << openBr
+               << "исследуемое отношение" << closeBr << newline << beginToDo << newline << GetFirstParameter(params)
+               << newline << endToDo << newline << texSlash << "end" << openBr << "scnhaselementrolelist" << closeBr
+               << newline;
+        return result.str();
+    };
+};
+
+class ScnSdClassCommand : public ScSCnTexCommand
+{
+public:
+    std::string Complete(TexCommandParams const &params) override
+    {
+        std::stringstream result;
+        result << texSlash << "begin" << openBr << "scnhaselementrolelist" << closeBr << openBr
```

Comment:
So, such code is hard anyway. I propose to create template functions to customize styles of strings

Reply:
to optimize this code I will spend more time then this code will be used, this feature is for one time using, I think it don't need to be ideal optimize